notes/12-guarding-routes.md

```diff
@@ -0,0 +1,195 @@
+# Guarding Routes
+
+In this exercise, we will be adding redirection in the UI flow based on the availability of an authenticated session.
+Specifically, users should be able to access the `login` route, only when an authenticated session _does not exist_. And similarly, users should be able to access the `teams` route only when an authenticated _session exists_.
+
+So let's get started.
+
+## Changes Needed in Javascript
+
+First let's implement the auth service that we will be using to do session validation(or invalidation). This auth service is defined at `app/services/auth.js`.
```

Comment:
They've already been working with an auth service since step 10/11. Maybe change this to
```suggestion
First let's enhance the auth service so that we can easily tell if there's a logged in user, and implement a `logout` action. This auth service is defined at [`app/services/auth.js`](../app/services/auth.js). 
```